pkg/config/schema/schema.cue

```diff
@@ -0,0 +1,99 @@
+package config
+
+#icon: {
+    base64data: !=""
+    mediatype: !=""
+}
+
+// Generic gvk struct
+#gvk: {
+    group: !=""
+    version: !=""
+    kind: !=""
+}
+
+#olmgvkprovided: {
+  type: "olm.gvk.provided"
+  value: #gvk
+}
+
+#olmgvkrequired: {
+  type: "olm.gvk.required"
+  value: #gvk
+}
+
+// Generic package struct
+#package: {
+  packageName: !=""
+  version: !=""
+}
+
+#olmpackageprovided: {
+  type: "olm.package.provided"
+  value: #package
+}
+
+#olmpackagerequired: {
+  type: "olm.package.required"
+  value: #package
+}
+
+// Generic channel struct
+#channel: {
+  name: !=""
+  replaces?: !=""
+}
+
+#olmchannel: {
+  type: "olm.channel"
+  value: #channel
+}
+
+#olmskips: {
+  type: "olm.skips"
+  value: !=""
+}
+
+#olmskipRange: {
+  type: "olm.skipRange"
+  value: !=""
+}
+
+// Generic property struct
+#property: {
+  type: !=""
+  ...
+}
+
+#relatedImages: {
+  name: !=""
+  image: !=""
+}
+
+// schema: "olm.package"
+#olmpackage: {
+  schema: "olm.package"
+  name: !=""
+  defaultChannel: !=""
+  icon: #icon
+  description: !=""
+  properties?: [...#property]
+  ...
+}
+
+// schema: "olm.bundle"
+#olmbundle: {
+  schema: "olm.bundle"
+  name: !=""
```

Comment:
a lot of these `!=""` should probably specify types as well, this should be a string too

Reply:
I think `!=""` only works with string. Plus, I'm still having problem with json validation library in cuelang doesn't honor this non-empty check.

Reply:
I opened a PR on cue repo to address this bug: https://github.com/cuelang/cue/pull/853